lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -1042,6 +1040,7 @@ def mock_shutil_which(
         with monkeypatch.context() as m:
             m.setattr(shutil, "which", mock_shutil_which)
             m.setattr(subprocess, "Popen", MockPopen)
+            m.setattr(Inventory, "close", MockBufferedReader.close)
```

Comment:
I suppose that's fine that you're mocking `Inventory.close` to a different `close` you also don't expect to be called, and you cover all of the `Inventory` class below, but it looks odd and the self-documentation here (plus the assert if it *is* called) could be better. Ah, well.

Reply:
It _should_ be fine to mock `Inventory.close()` with a different `close()`, given that they are all supposed to be providing the same `io.IOBase.close()` interface!  😆  (And, yeah, since I expect none of them to be called, using separate implementations seems silly and redundant.  😉)

Nevertheless, I don't disagree that, on the surface, it looks funny.  🙂

As for the documentation...well, Nick is still looking at the test code, so there is still a chance to fix that.  🙃  What in particular would you like to see?

Reply:
> once you start to write that and think about how it'll look, you'll probably just write the "proper" mock instead.

One of the problems that I'm trying to address is to minimize the amount of code in the unit tests which is **not** executed per the coverage report.  So, adding an additional mock which causes the test to fail if it is called actually moves away from that goal.

> the clearest documentation would be a mocked `Inventory.close` that doesn't claim to be a stream close.

I find this to be curious, because, AIUI, we want `Inventory` to act like a stream, right?  (I mean, it provides all the same methods with basically the same semantics.)  So, it makes sense to me to use a stream-close mock for `Inventory.close()` (in this weird instance, anyway)....

>  Second best would be a comment that we're just being lazy here because it doesn't really matter as we don't expect it to be called.

I'll do that, if you think it's worth it (although, I don't think I'll term it as "lazy", because that's not really what it's accomplishing for me... 🙂).

Reply:
You should not wait for me to check the test code before you merge. I don't know how long it will take me :grin:

Reply:
> I find this to be curious, because, AIUI, we want `Inventory` to act like a stream, right? (I mean, it provides all the same methods with basically the same semantics.) So, it makes sense to me to use a stream-close mock for `Inventory.close()` (in this weird instance, anyway)....

`Inventory` *composes* a stream and a "stream provider" so they can be managed together. That composition is critical, and eliding that to mock it as if it *were* a stream just seems really obscure and awkward to me. But what you've got will work, even if it might confuse a future reader who even notices it. I approved, so I'll leave the rest to you.